src/transformers/models/realm/tokenization_realm.py

```diff
@@ -0,0 +1,146 @@
+# coding=utf-8
+# Copyright 2021 The REALM authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for REALM."""
+
+from ...file_utils import PaddingStrategy
+from ...tokenization_utils_base import BatchEncoding
+from ...utils import logging
+from ..bert.tokenization_bert import BertTokenizer
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "realm-cc-news-pretrained-embedder": "https://huggingface.co/qqaatw/realm-cc-news-pretrained-embedder/resolve/main/vocab.txt",
+        "realm-cc-news-pretrained-retriever": "https://huggingface.co/qqaatw/realm-cc-news-pretrained-retriever/resolve/main/vocab.txt",
+        "realm-cc-news-pretrained-encoder": "https://huggingface.co/qqaatw/realm-cc-news-pretrained-encoder/resolve/main/vocab.txt",
+        "realm-orqa-nq-searcher": "https://huggingface.co/qqaatw/realm-orqa-nq-searcher/resolve/main/vocab.txt",
+        "realm-orqa-nq-reader": "https://huggingface.co/qqaatw/realm-orqa-nq-searcher/resolve/main/vocab.txt",
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "realm-cc-news-pretrained-embedder": 512,
+    "realm-cc-news-pretrained-retriever": 512,
+    "realm-cc-news-pretrained-encoder": 512,
+    "realm-orqa-nq-searcher": 512,
+    "realm-orqa-nq-reader": 512,
+}
+
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "realm-cc-news-pretrained-embedder": {"do_lower_case": True},
+    "realm-cc-news-pretrained-retriever": {"do_lower_case": True},
+    "realm-cc-news-pretrained-encoder": {"do_lower_case": True},
+    "realm-orqa-nq-searcher": {"do_lower_case": True},
+    "realm-orqa-nq-reader": {"do_lower_case": True},
+}
+
+
+class RealmTokenizer(BertTokenizer):
```

Comment:
The tokenizer file already looks very nice! Could you however please remove the abstraction of `BertTokenizer` and instead just copy-paste the code?

Reply:
I can also take care of it if you want :-)

Reply:
I see there are many models which directly subclass `BertTokenizer` so I think this way is probably acceptable in `transformers` and leads to a more concise tokenization file?  I'm fine to copy-paste the code if this abstraction is no longer acceptable by HF team.
```
17 results - 15 files

src/transformers/models/bert_japanese/tokenization_bert_japanese.py:
  71  
  72: class BertJapaneseTokenizer(BertTokenizer):
  73      r"""

src/transformers/models/convbert/tokenization_convbert.py:
  45  
  46: class ConvBertTokenizer(BertTokenizer):
  47      r"""

src/transformers/models/distilbert/tokenization_distilbert.py:
  55  
  56: class DistilBertTokenizer(BertTokenizer):
  57      r"""

src/transformers/models/dpr/tokenization_dpr.py:
   89  
   90: class DPRContextEncoderTokenizer(BertTokenizer):
   91      r"""

  105  
  106: class DPRQuestionEncoderTokenizer(BertTokenizer):
  107      r"""

src/transformers/models/electra/tokenization_electra.py:
  51  
  52: class ElectraTokenizer(BertTokenizer):
  53      r"""

src/transformers/models/funnel/tokenization_funnel.py:
  57  
  58: class FunnelTokenizer(BertTokenizer):
  59      r"""

src/transformers/models/layoutlm/tokenization_layoutlm.py:
  45  
  46: class LayoutLMTokenizer(BertTokenizer):
  47      r"""

src/transformers/models/lxmert/tokenization_lxmert.py:
  35  
  36: class LxmertTokenizer(BertTokenizer):
  37      r"""

src/transformers/models/mobilebert/tokenization_mobilebert.py:
  35  
  36: class MobileBertTokenizer(BertTokenizer):
  37      r"""

src/transformers/models/realm/tokenization_realm.py:
  54  
  55: class RealmTokenizer(BertTokenizer):
  56      r"""

src/transformers/models/retribert/tokenization_retribert.py:
  40  
  41: class RetriBertTokenizer(BertTokenizer):
  42      r"""

src/transformers/models/squeezebert/tokenization_squeezebert.py:
  46  
  47: class SqueezeBertTokenizer(BertTokenizer):
  48      r"""

templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py:
  41  
  42: class {{cookiecutter.camelcase_modelname}}Tokenizer(BertTokenizer):
  43      r"""

tests/test_tokenization_auto.py:
  58  
  59: class NewTokenizer(BertTokenizer):
  60      pass

tests/test_tokenization_common.py:
  3585  
  3586: class FakeTokenizer(BertTokenizer):
  3587      pass

  3599  
  3600: class FakeTokenizer(BertTokenizer):
  3601      pass

```

Reply:
Very good observation! It's very true that it is not consistent to favor having stand-alone tokenizers while using an abstraction in the tokenizer templates... IMO the goal should be to break up all of the above mentioned abstractions that you've listed. I would be ok to leave it in this PR as it is, but IMO we should bump the priority of breaking up those tokenizers a bit cc @sgugger @LysandreJik @patil-suraj  - what do you think?

Reply:
Very much agree with Patrick here.

Reply:
Agreed too!

Reply:
Will do along with the PR mentioned here 
https://github.com/huggingface/transformers/pull/13292#discussion_r780749863

Reply:
Resolved in #15211.